presentation/src/main/java/com/mashup/gabbangzip/sharedalbum/presentation/utils/SwipeExt.kt

```diff
@@ -0,0 +1,97 @@
+package com.mashup.gabbangzip.sharedalbum.presentation.utils
+
+import android.annotation.SuppressLint
+import androidx.compose.animation.core.Animatable
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.gestures.detectDragGestures
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.input.pointer.positionChange
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlin.math.abs
+
+@Composable
+fun rememberSwipeState(maxWidth: Float, maxHeight: Float): Swipe =
+    remember { Swipe(maxWidth, maxHeight) }
+
+open class Swipe(val maxWidth: Float, val maxHeight: Float) {
+    val offsetX = Animatable(0f)
+    val offsetY = Animatable(0f)
+
+    fun reset(scope: CoroutineScope) = scope.launch {
+        launch { offsetX.animateTo(0f, tween(400)) }
+        launch { offsetY.animateTo(0f, tween(400)) }
+    }
+
+    fun liked(scope: CoroutineScope) = scope.launch {
+        offsetX.animateTo(maxWidth * 2, tween(400))
+    }
+
+    fun disLiked(scope: CoroutineScope) = scope.launch {
+        offsetX.animateTo(-(maxWidth * 2), tween(400))
+    }
+
+    fun drag(scope: CoroutineScope, x: Float, y: Float) = scope.launch {
+        launch { offsetX.animateTo(x) }
+        launch { offsetY.animateTo(y) }
+    }
+}
+
+@SuppressLint("ModifierFactoryUnreferencedReceiver")
+fun Modifier.swiper(
+    state: Swipe,
+    onDragReset: () -> Unit = {},
+    onDragLiked: () -> Unit,
+    onDragDisLiked: () -> Unit,
+): Modifier = composed {
+    val scope = rememberCoroutineScope()
+    Modifier
```

Comment:
```suggestion
fun Modifier.swiper(
    state: Swipe,
    onDragReset: () -> Unit = {},
    onDragLiked: () -> Unit,
    onDragDisLiked: () -> Unit,
): Modifier = composed {
    val scope = rememberCoroutineScope()
    this then Modifier
```
ModifierFactoryUnreferencedReceiver 가 Modifier 의 확장함수인데 앞에서 있던 Modifier 속성들을 유지하지 않아서 생기는 문제로 보이는 것 같아!
SuppressLint 로 무시하기보다는 위와 같이 this then 을 통해 전달된 Modifier 속성을 유지해보는 것은 어떨까?

Reply:
아아 이거 너무 좋습니다.... 경고가 뜨는게 불편해서 그냥 auto로 어노테이션으로 해결하긴 했는데 당연히 그런 어노테이션은 지양하는게 나도 좋다는 생각이야..! 계원이가 제안해준 코드로 수정할게!